BLUESPAWN-win-client/src/user/BLUESPAWN.cpp

```diff
@@ -0,0 +1,459 @@
+            if(result["debug"].as<int>() >= 1) {
+                Log::LogLevel::LogVerbose1.Enable();
+            }
+            if(result["debug"].as<int>() >= 2) {
+                Log::LogLevel::LogVerbose2.Enable();
+            }
+            if(result["debug"].as<int>() >= 3) {
+                Log::LogLevel::LogVerbose3.Enable();
+            }
+        }
+
+        ParseLogSinks(result["log"].as<std::string>());
+
+        if(result.count("hunt") || result.count("monitor")) {
+            if(result.count("hunt")) {
+                bluespawn.EnableMode(BluespawnMode::HUNT);
+            }
+            if(result.count("monitor")) {
+                bluespawn.EnableMode(BluespawnMode::MONITOR);
+            }
+
+            if(result.count("aggressiveness")) {
+                bluespawn.EnableMode(BluespawnMode::SCAN, static_cast<DWORD>(GetAggressiveness(result["aggressivenes"
+                                                                                                      "s"])));
+            }
+
+            if(result.count("hunts")) {
+                bluespawn.SetIncludedHunts(result["hunts"].as<std::vector<std::string>>());
+            } else if(result.count("exclude-hunts")) {
+                bluespawn.SetExcludedHunts(result["exclude-hunts"].as<std::vector<std::string>>());
+            }
+
+            auto UserReactions = result["react"].as<std::string>();
+            std::set<std::string> reaction_set;
+            for(unsigned startIdx = 0; startIdx < UserReactions.size();) {
+                auto endIdx{ UserReactions.find(',', startIdx) };
+                auto sink{ UserReactions.substr(startIdx, endIdx - startIdx) };
+                reaction_set.emplace(sink);
+                startIdx = endIdx + 1;
+                if(endIdx == std::string::npos) {
+                    break;
+                }
+            }
+
+            for(auto reaction : reaction_set) {
+                if(reactions.find(reaction) != reactions.end()) {
+                    bluespawn.AddReaction(std::move(reactions[reaction]));
+                } else {
+                    bluespawn.io.AlertUser(L"Unknown reaction \"" + StringToWidestring(reaction) + L"\"", INFINITY,
+                                           ImportanceLevel::MEDIUM);
+                }
+            }
+        }
+
+        if(result.count("mitigate")) {
+            bool bForceEnforce = false;
+            if(result.count("force"))
+                bForceEnforce = true;
+
+            MitigationMode mode = MitigationMode::Audit;
+            if(result["action"].as<std::string>() == "e" || result["action"].as<std::string>() == "enforce")
+                mode = MitigationMode::Enforce;
+
+            bluespawn.EnableMode(BluespawnMode::MITIGATE,
+                                 (static_cast<DWORD>(bForceEnforce) << 1) | (static_cast<DWORD>(mode) << 0));
+        }
+
+        bluespawn.Run();
+    } catch(cxxopts::OptionParseException e1) {
```

Comment:
Codacy found an issue: [Exception should be caught by reference.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5982821)